### DISCUSSION TITLE
Q: Skip specific files when using in-source tests

### DISCUSSION THREAD

**Discussion by jambonrose** [vitest-dev/vitest#3658] [Q&A] [answered]
I'm new to vitest and enjoying the experience so far. Thank you for all the hard work!

I'm using the [in-source testing feature](https://vitest.dev/guide/in-source.html) to colocate tests and code in the same file. By default, if a file doesn't have tests, Vitest outputs:

```
 FAIL  path/to/file.ts [ path/to/file.ts ]
Error: No test suite found in file /absolute/path/to/file.ts
```

This default behavior makes sense to me and helps ensure that I am intentional about writing tests for code. 

However, there are some files which I wish to omit from the testing framework. For example, some `index.ts` files merely collect functions/variables declared in other files, or sometimes files simply declare variables or initialize code. My question is: is there a recommended way to tell Vite to skip these files so that I am not shown failures when said files are intentionally missing a test suite?

I was looking in the documentation for some comment or directive that I would be able to add to a file (something like `/* vitest:skip */`) but I did not see any. Instead, I am using the `vite.config.ts` config to exclude these files either by name (like `index.ts`) or else via a pattern (resulting in filenames like `client.no-test.ts`). Both of these feel a little clumsy, especially as I am in a position where no `index.ts` files can contain tests!

Is there a different recommended way to handle this?

**Comment by sheremet-va** [comment, marked as answer]
Do you use `includeSource` field? It doesn't run tests if you don't have `import.meta.vitest` inside. Don't use `include` for your source code, it's only for test _files_.

**Comment by jambonrose** [comment]
Thank you for the quick response! You nailed it: I'm using `include` instead of `includeSource`.

I admit I'm a little disappointed that Vitest won't warn me when new files need tests, but I recognize that the choice is not clear-cut.

Thanks for your help!
